getdata_HN.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+import json
+import os.path
+import requests
+from urllib.parse import urljoin
+
+URL = 'https://hacker-news.firebaseio.com/v0/'
+KEYWORDS = ['python', 'django']
+DATAFILE = 'data.json'
+stories = {}
+
+
+def _get_json(param):
+    url = urljoin(URL, param)
+    res = requests.get(url)
+    return res.json()
+
+
+def _get_stories():
+    parameter = 'topstories.json?print=pretty'
+    return _get_json(parameter)
+    '''
+    parameter = 'topstories.json?print=pretty'
+    story_ids = []
+    for story_id in _get_json(parameter):
+        if len(story_ids) == 150:
+            break
+        else:
+            story_ids.append(story_id)
+    return story_ids
+    '''
+
+
+def _get_stories_id():
+    stories_id = []
+    for story in _get_stories():
+        parameter = 'item/{}.json?print=pretty'.format(story)
+        story = _get_json(parameter)
+        for kwd in KEYWORDS:
+            if (kwd in story['title']
+                ) or (kwd.upper() in story['title']
+                      ) or (kwd.capitalize() in story['title']):
+                stories_id.append(story['id'])
+    return stories_id
+
+
+def _add_stories(new_stories):
+    for story_id in new_stories:
+        parameter = 'item/{}.json?print=pretty'.format(story_id)
+        story = _get_json(parameter)
+        if 'url' in story:
+            url_story = story['url']
+        else:
+            url_story = '#'
+        stories[story['id']] = {
+            'title': story['title'],
+            'url': url_story
+        }
+    return
+
+
+def _save_stories(all_stories):
+    with open(DATAFILE, 'w') as outfile:
+        json.dump(all_stories, outfile, indent=4)
+    return
+
+
+def main():
+    if os.path.isfile(DATAFILE):
+        new_stories = {}
+        with open(DATAFILE) as json_data:
+            old_stories = json.load(json_data)
+            old_ids = old_stories.keys()
+            new_stories = old_stories
+            if _get_stories_id():
+                for new_story_id in _get_stories_id():
+                    for old_story_id in list(old_ids):
+                        if new_story_id != old_story_id:
```

Comment:
you are comparing two dictionary and then add new one to old one, or say in other words, you are merging two dict.

```
d1.update(d2)
```

seems enough